venues/ICLR.cc/2020/Conference/python/buddy_ac.py

```diff
@@ -0,0 +1,136 @@
+
+    if not buddy_ac_parent_group:
+        ac_group = client.get_group(conference.get_area_chairs_id())
+
+        buddy_ac_group = client.post_group(
+            openreview.Group(
+                id = conference_id + '/' + buddy_ac_parent_group_name,
+                readers = [
+                    conference_id,
+                    conference.get_program_chairs_id(),
+                    conference.get_area_chairs_id()
+                ],
+                signatories = [conference_id + '/' + buddy_ac_parent_group_name],
+                signatures = [conference_id],
+                writers = [conference_id],
+                web = 'buddyAreachairWebfield.js',
+                members = ac_group.members
+            )
+        )
+
+    map_meta_review_invitations = {invitation.id.split('Paper')[1].split('/')[0]: invitation for invitation in openreview.tools.iterget_invitations(client, regex = conference_id + '/Paper[0-9]+/-/Meta_Review')}
+
+    map_paper_ac_parent_groups = {group.id.split('Paper')[1].split('/')[0]: group for group in openreview.tools.iterget_groups(client, regex = conference_id + '/Paper[0-9]+/Area_Chairs$')}
+
+    map_paper_ac_individual_groups = {group.id.split('Paper')[1].split('/')[0]: group for group in openreview.tools.iterget_groups(client, regex = conference_id + '/Paper[0-9]+/Area_Chair1$')}
+
+    map_paper_buddy_ac_groups = {group.id.split('Paper')[1].split('/')[0]: group for group in openreview.tools.iterget_groups(client, regex = conference_id + '/Paper[0-9]+/' + buddy_ac_individual_group_name + '$')}
+
+    submissions = conference.get_submissions()
+    for paper in submissions:
+        paper_number = str(paper.number)
+        individual_buddy_group_id = conference_id + '/Paper' + paper_number + '/' + buddy_ac_individual_group_name
+
+        ## Create buddy ac group for this paper
+        if paper_number not in map_paper_buddy_ac_groups:
+            individual_buddy_group = client.post_group(
+                openreview.Group(
+                    id = individual_buddy_group_id,
+                    readers = [
+                        conference_id,
+                        conference.get_program_chairs_id(),
+                        conference.get_area_chairs_id(paper_number),
+                        individual_buddy_group_id],
+                    nonreaders = [conference_id + '/Paper' + paper_number + '/Authors'],
+                    signatories = [individual_buddy_group_id],
+                    signatures = [conference.get_id()],
+                    writers = [conference.get_id()],
+                    members = []
+                )
+            )
+
+        paper_ac_individual_group = map_paper_ac_individual_groups.get(paper_number)
+        if paper_ac_individual_group and (individual_buddy_group_id not in paper_ac_individual_group.readers):
+            paper_ac_individual_group.readers.append(individual_buddy_group_id)
```

Comment:
we need to do this to reveal the AC identity to the Buddy AC?

Reply:
Yeah, both the buddy and the primary AC should be able to see each other's name.